subsys/bluetooth/host/hci_core.c

```diff
@@ -6718,7 +6718,9 @@ void bt_id_get(bt_addr_le_t *addrs, size_t *count)
 {
 	size_t to_copy = MIN(*count, bt_dev.id_count);
 
-	memcpy(addrs, bt_dev.id_addr, to_copy * sizeof(bt_addr_le_t));
+	if (addrs) {
+		memcpy(addrs, bt_dev.id_addr, to_copy * sizeof(bt_addr_le_t));
+	}
 	*count = to_copy;
```

Comment:
Should we move the MIN to only be done when addrs pointer is given?
```
size_t count;
bt_id_get(NULL, &count);
printk("Number of identities %zu\n", count);
```
Would make sense to do something like the above, without initializing count to something.

Reply:
Sure, no objections to that

Reply:
Or, just a new bt_id_get_count() since this doesn’t have much to do with the purpose of the original API anymore

Reply:
Updated.